src/frontend/src/index.ts

```diff
@@ -104,17 +105,24 @@ const init = async () => {
   const addDeviceAnchor = getAddDeviceAnchor();
   if (addDeviceAnchor !== undefined) {
     // Register this device (tentatively)
-    await registerTentativeDevice(addDeviceAnchor, connection);
-
-    const i18n = new I18n();
-    const copy = i18n.i18n(copyJson);
-
-    // Show a good bye message
-    showMessage({
-      message: copy.close_page_device_added,
-      role: "notify-device-added",
-    });
-    return;
+    const { userNumber, alias: deviceAlias } = await registerTentativeDevice(
+      addDeviceAnchor,
+      connection
+    );
+
+    // Display a success page once device added (above registerTentativeDevice **never** returns if it fails)
+    await addDeviceSuccess({ deviceAlias });
+
+    // If user "Click" continue in success page, proceed with authentication
+    const result = await authenticate(connection, userNumber);
+    const { success } = await handleLoginFlowResult(result);
+
+    // User have successfully signed-in we can jump to manage page
+    if (nonNullish(success)) {
```

Comment:
what does the user see if it's not a success?

Reply:
There was an error which was prompted or user clicked "Cancel" to which he/she except that nothing else happens.